### PR TITLE
Update pure graalpy examples

### PR DESCRIPTION
**Pull request by bhoran**
Update the GraalPy examples so that they use includes. The descriptions are still a little wooden, so any help appreciated!

### CODE REVIEW

**Comment by timfel** [comment]
Looks good. The link to the examples seems broken, I think we need this change:
```diff
modified   docs/graalpy/guides/GraalPy.md
@@ -86,4 +86,4 @@ A GraalPy option that comes with a JVM is identified by the suffix "-jvm", for e
 
 ---
 
-## [GraalPy Examples](/graalpy/examples/){:target="_blank"}
+## [GraalPy Examples](examples)
```

File: docs/graalpy/examples/includes/step_unset_java-home.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Before you create a native executable, unset your `JAVA_HOME` environment variable. 
+(For example, on Linux and macOS, use the command `unset JAVA_HOME`.)
```
Comment (timfel) [inline docs/graalpy/examples/includes/step_unset_java-home.md:2]:
```suggestion
(On Linux and macOS, use the command `unset JAVA_HOME`. In Windows' PowerShell, use `$Env:JAVA_HOME=''`)
```

**Review by timfel — APPROVED** [review]

File: docs/graalpy/examples/BeautifulSoup_Application.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+title: BeautifulSoup Application
+description: This example shows how to use the [`requests`](https://pypi.org/project/requests/) and [`beautifulsoup`](https://www.crummy.com/software/BeautifulSoup/) modules to scrape a web page.  GraalPy is then used to create a native executable.
+filename: currency_exchange.py
```
Comment (olyagpl) [inline docs/graalpy/examples/BeautifulSoup_Application.md:4]: Where is the permalink value for these example pages? How do we specify it?

**Review by olyagpl — APPROVED** [review]

File: docs/graalpy/examples/BeautifulSoup_Application.md (hunk 2 above)
Comment (bhoran) [reply docs/graalpy/examples/BeautifulSoup_Application.md:4]: Oh, good question. I'll fix that.

**Pull request merged**